### PR TITLE
Bump sonata block bundle dependency

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because the block methods have changed.

## Changelog

```markdown
### Changed
- Use `sonata-project/block-bundle` 4
```

### CODE REVIEW

**Comment by OskarStark** [comment]
Not possible against stable branch?

File: src/Block/FormatterBlockService.php [hunk 1]
```diff
@@ -37,7 +40,7 @@ public function execute(BlockContextInterface $blockContext, Response $response
         ], $response);
     }
 
-    public function buildEditForm(FormMapper $formMapper, BlockInterface $block): void
+    public function configureEditForm(FormMapper $form, BlockInterface $block): void
```
Comment (core23) [inline src/Block/FormatterBlockService.php]: The API has changed here @OskarStark

File: composer.json [hunk 2]
```diff
@@ -23,8 +23,8 @@
     "require": {
         "php": "^7.1",
         "psr/log": "^1.0",
-        "sonata-project/block-bundle": "^3.11",
-        "sonata-project/core-bundle": "^3.9",
+        "sonata-project/block-bundle": "^4.0",
```
Comment (phansys) [inline composer.json]: We should stop building with PHP < 7.3 based on this constraint ([ref](https://travis-ci.org/sonata-project/SonataFormatterBundle/jobs/622760538#L382)).

File: composer.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
         "twig/twig": "^2.4"
     },
     "require-dev": {
-        "matthiasnoback/symfony-dependency-injection-test": "^3.0",
+        "matthiasnoback/symfony-dependency-injection-test": "^4.0",
         "symfony/phpunit-bridge": "^4.3"
```
Comment (franmomu) [inline composer.json]: We can bump this to ^5.0 as well

**Pull request closed without merging**